### PR TITLE
Docker: add picolibc-include directory

### LINKED ISSUE #2205: docker: latest dev image missing picolibc.h?
The CI build for the TinyGo Drivers repo is currently failing. It appears that there is something missing from the Docker image related to the recent PR merged related to this.
 
```
tinygo build -size short -o ./build/test.hex -target=itsybitsy-m0 ./examples/adt7410/main.go
In file included from /tinygo/lib/picolibc-stdio.c:4:
In file included from /tinygo/lib/picolibc/newlib/libc/tinystdio/stdio.h:44:
In file included from /usr/lib/llvm-11/lib/clang/11.1.0/include/inttypes.h:21:
In file included from /tinygo/lib/picolibc/newlib/libc/include/inttypes.h:16:
/tinygo/lib/picolibc/newlib/libc/include/newlib.h:35:10: fatal error: 'picolibc.h' file not found
#include <picolibc.h>
         ^~~~~~~~~~~~
```
See https://app.circleci.com/pipelines/github/tinygo-org/drivers/988/workflows/46cbb77e-c7e0-4437-8ef7-98c45dd3da6e/jobs/1014?invite=true#step-105-8

### PR DESCRIPTION
**Pull request by ysoldak**
Fixes #2205 

Verified locally by building docker image and running `make smoke-test` in tinygo-drivers.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -16,6 +16,7 @@ RUN cd /tinygo/ && \
     git submodule update --init --recursive --force
 
 COPY ./lib/picolibc-* /tinygo/lib/
+COPY ./lib/picolibc-include/* /tinygo/lib/picolibc-include/
```
Comment (aykevl) [inline Dockerfile:19]: Huh, why does the `COPY ./lib/picolibc-* /tinygo/lib/` not work?
The goal is to copy both `./lib/picolibc-include` and `./lib/picolibc-stdio.c`. So the PR as-is is incomplete: the `./lib/picolibc-stdio.c` file also needs to be copied.

**Comment by aykevl** [comment]
(Sidenote: #2110 will actually remove `./lib/picolibc-include/`)

File: Dockerfile (hunk 1 above)
Comment (ysoldak) [reply Dockerfile:19]: I don't know. This is how COPY works. I've verified multiple times, with * it only copies files, not dirs. So `picolibc-stdio.c` is copied alright by the line 18, and for the dir we need line 19.

**Comment by ysoldak** [comment]
@aykevl you plan to merge #2110 first so this PR is obsolete or we merge this to unblock drivers?

File: Dockerfile (hunk 1 above)
Comment (deadprogram) [reply Dockerfile:19]: The Dockerfile `COPY` command is not completely intuitive as far as handling wildcards and directories. :crying_cat_face: 

In any case, merging to fix this issue. Thanks @ysoldak

**Pull request merged**